src/svgdevicecontext.cpp

```diff
@@ -163,6 +163,11 @@ void SvgDeviceContext::StartGraphic(Object *object, std::string gClass, std::str
         }
     }
 
+    // make sure we are in a g container
+    while (strcmp(m_currentNode.name(), "g")) {
+        m_currentNode = m_currentNode.parent();
+    }
```

Comment:
I am not convinced this is the way to go. Won't this change the hierarchy in the SVG? I have to have a closer look but probably the only thing to do is to call Start/EndTextGraphic when calling from DrawTextEditorialElement and Start/EndGraphic otherwise. This can be determined with an additional parameter in DrawAnnot. Wouldn't that work?